src/index.ts

```diff
@@ -1,32 +1,82 @@
 import SERVICES from './services';
 import './index.css';
 import { debounce } from 'debounce';
+import { API , PatternPasteEventDetail } from '@editorjs/editorjs';
 
 /**
- * @typedef {object} EmbedData
  * @description Embed Tool data
- * @property {string} service - service name
- * @property {string} url - source URL of embedded content
- * @property {string} embed - URL to source embed page
- * @property {number} [width] - embedded content width
- * @property {number} [height] - embedded content height
- * @property {string} [caption] - content caption
  */
+interface EmbedData {
+  /** Service name */
+  service: string;
+  /** Source URL of embedded content */
+  source: string;
+  /** URL to source embed page */
+  embed: string;
+  /** Embedded content width */
+  width?: number;
+  /** Embedded content height */
+  height?: number;
+  /** Content caption */
+  caption?: string;
+}
+
 /**
- * @typedef {object} PasteEvent
- * @typedef {object} HTMLElement
- * @typedef {object} Service
  * @description Service configuration object
- * @property {RegExp} regex - pattern of source URLs
- * @property {string} embedUrl - URL scheme to embedded page. Use '<%= remote_id %>' to define a place to insert resource id
- * @property {string} html - iframe which contains embedded content
- * @property {Function} [id] - function to get resource id from RegExp groups
  */
+interface Service {
+  /** Pattern of source URLs */
+  regex: RegExp;
+  /** URL scheme to embedded page. Use '<%= remote_id %>' to define a place to insert resource id */
+  embedUrl: string;
+  /** Iframe which contains embedded content */
+  html: string;
+  /** Function to get resource id from RegExp groups */
+  id?: (ids: string[]) => string;
+  /** Embedded content width */
+  width?: number;
+  /** Embedded content height */
+  height?: number;
+}
+
 /**
- * @typedef {object} EmbedConfig
  * @description Embed tool configuration object
- * @property {object} [services] - additional services provided by user. Each property should contain Service object
  */
+interface EmbedConfig {
+  /** Additional services provided by user */
+  services?: { [key: string]: Service | boolean };
```

Comment:
maybe we should move `{ [key: string]: Service | boolean }` to some type since it is used in many places